ui/src/app/pages/homepage/data-access-requirements.tsx

```diff
@@ -0,0 +1,531 @@
+
+  // which module are we currently guiding the user to complete?
+  const [activeModule, setActiveModule] = useState(null);
+
+  const [navigate, ] = useNavigation();
+  const enabledModules = allModules.map(module => {
+    const enabledTaskMaybe = getRegistrationTask(navigate, module);
+    return enabledTaskMaybe && enabledTaskMaybe.module;
+  });
+
+  // whenever the profile changes, setActiveModule(the first incomplete enabled module)
+  useEffect(() => {
+    fp.flow(
+      fp.find<AccessModule>(module => {
+        const status = getAccessModuleStatusByName(profile, module);
+        return !bypassedOrCompletedText(status);
+      }),
+      setActiveModule
+    )
+    (enabledModules);
+  }, [profile]);
+
+  const DARHeader = () => <FlexColumn style={styles.headerFlexColumn}>
+    <Header style={styles.headerRW}>Researcher Workbench</Header>
+    <Header style={styles.headerDAR}>Data Access Requirements</Header>
+  </FlexColumn>;
+
+  const Completed = () => <FlexRow data-test-id='dar-completed' style={styles.completed}>
+    <FlexColumn>
+      <div style={styles.completedHeader}>Thank you for completing all the necessary steps</div>
+      <div style={styles.completedText}>Researcher Workbench data access is complete.</div>
+    </FlexColumn>
+    <GetStartedButton style={{marginLeft: 'auto'}}/>
+  </FlexRow>;
+
+  const SelfBypass = () => {
+    const selfBypass = async() => {
+      spinnerProps.showSpinner();
+      await bypassAll(allModules, true);
+      spinnerProps.hideSpinner();
+      reload();
+    };
+
+    return <FlexRow data-test-id='self-bypass' style={styles.selfBypass}>
+      <div style={styles.selfBypassText}>[Test environment] Self-service bypass is enabled</div>
+      <Button style={{marginLeft: '0.5rem'}} onClick={() => selfBypass()}>Bypass all</Button>
+    </FlexRow>;
+  };
+
+  const ModulesForCard = (props: {modules: AccessModule[]}) => {
+    const {modules} = props;
+    return <FlexColumn style={styles.modulesContainer}>
+      {modules.map(module =>
+          <MaybeModule key={module} module={module} active={module === activeModule}/>
+      )}
+    </FlexColumn>;
+  };
+
+  const RegisteredTierCard = () => {
+    return <FlexRow style={styles.card}>
+      <FlexColumn>
+        <div style={styles.cardStep}>Step 1</div>
+        <div style={styles.cardHeader}>Complete Registration</div>
+        <FlexRow>
+          <RegisteredTierBadge/>
+          <div style={styles.rtData}>Registered Tier data</div>
+        </FlexRow>
+        <div style={styles.rtDataDetails}>Once registered, you’ll have access to:</div>
+        <FlexRow>
+          <DARIcons.individual style={styles.rtDetailsIcon}/>
+          <div style={styles.rtDataDetails}>Individual (not aggregated) data</div>
+        </FlexRow>
+        <FlexRow>
+          <DARIcons.identifying style={styles.rtDetailsIcon}/>
+          <div style={styles.rtDataDetails}>Identifying information removed</div>
+        </FlexRow>
+        <FlexRow>
```

Comment:
nit: pattern is repeated a few times, seems common thread is a label and an icon. Could this also be a subcomponent? Or else generated dynamically from an array?

Reply:
Agreed.  I was looking for a better solution and wasn't able to find it.

I updated this with an OK solution, and I'll keep trying.

Reply:
this is now the DataDetail component.